### DISCUSSION TITLE
Hi i am trying to get user id!

### DISCUSSION THREAD

**Discussion by diogoribeirodev** [tweepy/tweepy#1676] [Questions] [answered]
Hi, i am trying to get a users id, but every time i call the get_user function i get this error TypeError: get_user() takes 1 positional argument but 2 were given and i dont understand why

auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)
api = tweepy.API(auth)
screen_name = "PracticeGfG"
user = api.get_user("PracticeGfG")
ID = user.id_str

print("The ID of the user is : " + ID)

**Comment by Harmon758** [comment, marked as answer]
[Tweepy v4.0.0](https://github.com/tweepy/tweepy/releases/tag/v4.0.0) stopped allowing positional arguments for `API.get_user` (the `1` in your error is `self`). This was because `id` was removed as an endpoint parameter for the method, since it's no longer a documented parameter for the endpoint, so it would be ambiguous whether the positional argument being passed is a user ID or screen name.

You can simply pass `screen_name` as a keyword argument.
See the [documentation for `API.get_user`](https://tweepy.readthedocs.io/en/stable/api.html#tweepy.API.get_user).

For code block usage, see https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks.
